hazelcast/src/main/java/com/hazelcast/nio/tcp/TcpIpConnectionManager.java

```diff
@@ -368,14 +364,7 @@ private TcpIpConnectionErrorHandler getErrorHandler(Address endpoint, boolean re
     public void onConnectionClose(Connection connection) {
         closedCount.inc();
 
-        if (activeConnections.remove(connection)) {
-            // this should not be needed; but some tests are using DroppingConnection which is not a TcpIpConnection.
```

Comment:
The EventLoopGroup will listen to channel closing, and takes actions accordingly.